src/constructors.jl

```diff
@@ -0,0 +1,172 @@
+abstract type AbstractHMCSampler <: AbstractMCMC.AbstractSampler end
+
+##############
+### Custom ###
+##############
+
+"""
+    HMCSampler
+
+A `AbstractMCMC.AbstractSampler` for kernels in AdvancedHMC.jl.
+
+# Fields
+
+$(FIELDS)
+
+# Notes
+
+Note that all the fields have the prefix `initial_` to indicate
+that these will not necessarily correspond to the `kernel`, `metric`,
+and `adaptor` after sampling.
+
+To access the updated fields use the resulting [`HMCState`](@ref).
+"""
+Base.@kwdef struct HMCSampler{
+    K<:AbstractMCMCKernel,
+    M<:AbstractMetric,
+    A<:Adaptation.AbstractAdaptor,
+} <: AbstractHMCSampler
+    "[`AbstractMCMCKernel`](@ref)."
+    kernel::K
+    "[`AbstractMetric`](@ref)."
+    metric::M
+    "[`AbstractAdaptor`](@ref)."
+    adaptor::A
+    "Adaptation steps if any"
+    n_adapts::Int = 0
+end
+
+############
+### NUTS ###
+############
+"""
+    NUTS(n_adapts::Int, δ::Real; max_depth::Int=10, Δ_max::Real=1000, init_ϵ::Real=0)
+
+No-U-Turn Sampler (NUTS) sampler.
+
+# Fields
+
+$(FIELDS)
+
+# Usage:
+
+```julia
+NUTS(n_adapts=1000, δ=0.65)  # Use 1000 adaption steps, and target accept ratio 0.65.
+```
+"""
+struct NUTS{T<:AbstractFloat,I,D} <: AbstractHMCSampler
+    "`n_adapts::Int` : Number of adaptation steps."
```

Comment:
Just
```suggestion
    "Number of adaptation steps."
```
should be sufficient. Similar suggestions apply to the other fields below 🙂